binarytree/__init__.py

```diff
@@ -649,6 +685,89 @@ def convert(bt):
     raise ValueError('Expecting a list or a node')
 
 
+def get_level(bt, level, show_values=False, show_nulls=False):
+    """Return the requested level of the binary tree, ordered from left to right.
+
+    If a node other than the root node is passed in, then this function
+    returns the requested level of the tree relative to the passed in node.
+
+    :param bt: The binary tree.
+    :type bt: binarytree.Node
+    :param level: The requested level to return.
+    :type level: int
+    :param show_values: whether to convert nodes to values before returning.
+    :type show_values: boolean
+    :param show_nulls: whether to show where empty nodes are.
+    :type show_nulls: boolean
+    :return: dictionary of form {0:[rootnode], 1:[1st, level, nodes]}
+    :rtype: dictionary
+    """
+    if not isinstance(bt, Node):
+        raise ValueError("Expecting a Node.")
+    if not isinstance(level, int) or level < 0:
+        raise ValueError("Requested level must be a non-negative integer.")
+    currlevel = 0
+    levels = {currlevel: [bt]}
+    while currlevel < level:
+        currlevel += 1
+        nodes = []
+        for node in levels[currlevel - 1]:
+            if node.left:
+                nodes.append(node.left)
+            elif show_nulls:
+                nodes.append(_create_node(_null))
+            if node.right:
+                nodes.append(node.right)
+            elif show_nulls:
+                nodes.append(_create_node(_null))
+        if len(nodes) == 0 or all(node.value is _null for node in nodes):
+            raise ValueError("Requested level not present in tree.")
+        levels[currlevel] = nodes
+    if show_values is True:
+        levels[level] = [node.value for node in levels[level]]
+    return levels[level]
+
+
+def get_levels(bt, show_values=False, show_nulls=False):
```

Comment:
I think returning a list of lists would be better here. Returning a dict with index as keys is computationally more expensive to create with no additional benefit.